File: tests/fixtures/codegen/expected/go/queries/admin/events/getevent.go
package events

import (
	"context"
	billing "example.com/pgsid-fixture/generated/go/schema/billing"
	public "example.com/pgsid-fixture/generated/go/schema/public"
)

const GetEventSQL = "SELECT id, source_id, state, metadata FROM billing.events WHERE id = $1;"

type GetEventParams struct {
	Id *int64 `db:"id"`
}
type GetEventRow struct {
	Id       billing.EventId        `db:"id"`
	SourceId public.EventId         `db:"source_id"`
	State    billing.EventState     `db:"state"`
	Metadata *billing.EventMetadata `db:"metadata"`
}

func (q *Queries) GetEvent(ctx context.Context, params GetEventParams) (GetEventRow, error) {
	var row GetEventRow
	err := q.db.QueryRow(ctx, GetEventSQL, params.Id).Scan(&row.Id, &row.SourceId, &row.State, &row.Metadata)
	return row, err
}

File: tests/fixtures/codegen/expected/typescript/runtime/queries/admin/events/GetEvent.ts
import type { GetEventParams, GetEventRow } from "../../../../types/queries/admin/events/GetEvent.js";
export type { GetEventParams, GetEventRow } from "../../../../types/queries/admin/events/GetEvent.js";
import type * as pgsid from "./pgsid/queryable.js";
export const getEventSql = "SELECT id, source_id, state, metadata FROM billing.events WHERE id = $1;";
export async function getEvent(db: pgsid.Queryable, params: GetEventParams): Promise<GetEventRow | undefined> {
    const result = await db.query(getEventSql, [params["id"]]);
    const row = result.rows[0];
    if (row === undefined)
        return undefined;
    return row as GetEventRow;
}
